api/client.go

```diff
@@ -1145,6 +1151,103 @@ func RequireState(states ...string) RequestCallback {
 	}
 }
 
+// CompareReplicationStates returns 1 if s1 is newer or identical, -1 if s1 is older, and 0
+// if neither s1 or s2 is strictly greater. An error is returned if s1 or s2
+// are invalid or from different clusters.
+func CompareReplicationStates(s1, s2 string) (int, error) {
+	w1, err := ParseRequiredState(s1, nil)
+	if err != nil {
+		return 0, err
+	}
+	w2, err := ParseRequiredState(s2, nil)
+	if err != nil {
+		return 0, err
+	}
+
+	if w1.ClusterID != w2.ClusterID {
+		return 0, fmt.Errorf("can't compare replication states with different ClusterIDs")
+	}
+
+	switch {
+	case w1.LocalIndex >= w2.LocalIndex && w1.ReplicatedIndex >= w2.ReplicatedIndex:
+		return 1, nil
+	// We've already handled the case where both are equal above, so really we're
+	// asking here if one or both are lesser.
+	case w1.LocalIndex <= w2.LocalIndex && w1.ReplicatedIndex <= w2.ReplicatedIndex:
+		return -1, nil
+	}
+
+	return 0, nil
+}
+
+func MergeStates(old []string, new string) []string {
```

Comment:
```suggestion
func MergeReplicationStates(old []string, new string) []string {
```

Please could you add a function documentation comment as well, similar to CompareReplicationStates?

Reply:
Updated